packages/twenty-front/src/modules/ui/field/input/components/DateInput.tsx

```diff
@@ -32,10 +30,13 @@ export type DateInputProps = {
   isDateTimeInput?: boolean;
   onClear?: () => void;
   onSubmit?: (newDate: Nullable<Date>) => void;
+  hideHeaderInput?: boolean;
+  temporaryValue: Nullable<Date>;
+  setTemporaryValue: (newValue: Nullable<Date>) => void;
+  wrapperRef: React.RefObject<HTMLElement>;
```

Comment:
can you make `temporaryValue`, `setTemporaryValue` and `wrapperRef` arguments optional? Then you can leave the instanciations if props if they are undefined. It will avoid to update `packages/twenty-front/src/modules/object-record/record-field/meta-types/input/components/DateTimeFieldInput.tsx`

Reply:
@charlesBochet wasn't satisfied with this code, and I asked him to take a look as he had an idea of the ideal API.

Reply:
I'm not convinced by the name _temporary value_, but I think it's not a bad idea to let parents provide `temporaryValue`, `setTemporaryValue`, and `wrapperRef`. It makes the component more easily customizable. I find it easier than making the arguments optional; I think it would make the child component less understandable—I don't think having an "optional" useState is common.

I can change my mind if we agree there is no better solution!